packages/xo-server-auth-ldap/src/index.js

```diff
@@ -418,11 +418,22 @@ class AuthLdap {
 
         const xoGroupMembers = xoGroup.users === undefined ? [] : xoGroup.users.slice(0)
 
-        for (const ldapId of ldapGroupMembers) {
-          const xoUser = xoUsers.find(user => user.authProviders.ldap.id === ldapId)
+        for (const memberId of ldapGroupMembers) {
+          const {
+            searchEntries: [ldapUser],
+          } = await client.search(this._searchBase, {
+            scope: 'sub',
+            filter: `(${escape(membersMapping.userAttribute)}=${escape(memberId)})`,
+          })
+          if (ldapUser === undefined) {
+            continue
+          }
+
+          const xoUser = xoUsers.find(user => user.authProviders.ldap.id === ldapUser[this._userIdAttribute])
           if (xoUser === undefined) {
             continue
           }
+
```

Comment:
Unrelated change, or maybe it's for make the comment more readable ?

Reply:
It was to better group `undefined` checks with their assignments, but I removed it.